### PR TITLE
Test Environment Setup

### PR DESCRIPTION
**Pull request by anuj-verma**
Added: Factory Girl, Faker

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -30,6 +30,9 @@ group :development, :test do
   gem 'web-console', '~> 2.0'
   gem 'spring'
   gem 'minitest'
+  gem "minitest-rails", "~> 2.2.0"
```
Comment (sivagollapalli) [inline Gemfile]: @anuj-verma I think we dont need `minitest-rails` gem for test env as we are rails testing framework. We have to include `minitest-rails` only if you want to use `spec` DSL. 

@jiren Am I correct?

Comment (jiren) [inline Gemfile]: Yes @sivagollapalli. Will remove it later. Added for generating tests files for existing models and controller.

File: test/controllers/users/omniauth_callbacks_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require 'test_helper'
+
+class Users::OmniauthCallbacksControllerTest < ActionController::TestCase
+
+  test "github signup" do
+    OmniAuth.config.add_mock(:github, {:uid => '12345'})
```
Comment (sivagollapalli) [inline test/controllers/users/omniauth_callbacks_controller_test.rb]: I think we can use `minitest-rspec_mocks` if we want mocking & stubbing. IMO, `rspec-mocks` provides great mocking features when we compare with `minitest`. 

Thoughts please?

Comment (jiren) [inline test/controllers/users/omniauth_callbacks_controller_test.rb]: Right now we have only one external API(Github) for signup/sign-in needs mocking. Github request mocking is provided by omniauth-github gem. So at this moment, there is no need of request mocking framework.

**Pull request merged**